### PR TITLE
Add foundations:elevation to storybook

### PR DESCRIPTION
**Pull request by jameskoster**
Screenshot 2024-10-15 at 14 48 15

Related to https://github.com/WordPress/gutenberg/pull/66122.

## What?
Adds a page detailing elevation as a foundation of the overall design language. 

## Why?
To help guide end users about how to use elevation when working with the WordPress components package, or when creating new UIs.

## Testing Instructions
1. `npm run storybook:dev`
2. Observe the new Foundations / Design language / Elevation section

## Missing pieces
This PR is a draft for now. I'd welcome assistance with adding:

1. Images – it's unclear if a Storybook addon is required for this, and where any images should be hosted.
2. Internal links – I tried adding the [links addon](https://storybook.js.org/addons/@storybook/addon-links) to create an internal link from the tokens section to the tokens page, but couldn't get it working.

### CODE REVIEW

File: storybook/stories/foundations/design-language/elevation.mdx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { Meta, Typeset } from '@storybook/addon-docs/blocks';
+import { linkTo, hrefTo } from '@storybook/addon-links';
+
+<Meta title="Foundations/Design Language/Elevation" name="page" />
+
+# Elevation
+
+Elevation, through the use of shadows, visually indicates the layers where overlapping UI elements reside. Shadows create the illusion of depth, showing how one UI element is positioned above another on the z-axis. Elevation should be used to organize elements, establish a clear hierarchy, and draw focus to key components.
+
+In the WordPress components package there are four levels of elevation:
```
Comment (ciampo) [inline storybook/stories/foundations/design-language/elevation.mdx]: Elevation (like every other tokens) is not part of the components package, but rather of the design system as a whole. Even technically, the elevation tokens are currently in `@wordpress/base-styles` — the components provided by the `@wordpress/components` package consume those tokens and therefore follow the design system.

In other words, from my perspective, the `@wordpress/components` package is a part of the design system's offering, as much as the token themselves.

**Comment by ciampo** [comment]
> This PR is a draft for now. I'd welcome assistance with adding:
> 
> 1. Images – it's unclear if a Storybook addon is required for this, and where any images should be hosted.
> 2. Internal links – I tried adding the [links addon](https://storybook.js.org/addons/@storybook/addon-links) to create an internal link from the tokens section to the tokens page, but couldn't get it working.

@mirka is this something you can help with, since you've been taking a look at other Storybook-related tasks?

File: storybook/stories/foundations/design-language/elevation.mdx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			<td>Large</td>
+			<td>For components that confirm or provide decisions or other interruption necessary tasks.</td>
+			<td>Modals</td>
+		</tr>
+	</tbody>
+</table>
+
+## Tokens
+
+Use tokens to apply elevation in your work. Please refer to the Tokens section of Storybook to learn more.
```
Comment (tyxla) [inline storybook/stories/foundations/design-language/elevation.mdx]: Let's make sure to add a link to the Elevation tokens page once it lands in #66122.

**Comment by ciampo** [comment]
Re. links, does [this section](https://storybook.js.org/docs/writing-docs/mdx#linking-to-other-stories-and-pages) of the docs help?

**Comment by mirka** [comment]
I pushed an example for adding images: ea10ad020c04b1aa6cb8a5476cd56b12c70b4af4

**Comment by jameskoster** [comment]
Thanks y'all, I added a link to the tokens section, obviously it will be broken until https://github.com/WordPress/gutenberg/pull/66122 is merged, so let's hold off on this one until then.

**Review by ciampo — APPROVED** [review]
Rebased to include the tokens page, and applied a few improvements / fixes.

Code-wise, this is LGTM 🚀 

I'll leave it to @WordPress/gutenberg-design to merge it when they're happy with the contents of the page.

**Comment by jameskoster** [comment]
Added an internal link from the tokens page to the foundation page. I think this is good as an initial iteration :)

**Comment by jameskoster** [comment]
Disabled auto-merge. I noticed a couple of copy tweaks to make, and I'm going to add a graphic to better demonstrate the examples :)

**Comment by ciampo** [comment]
Cool! I'll leave the merging to you :)

**Pull request merged**